test/system/030-run.bats

```diff
@@ -536,6 +536,17 @@ json-file | f
     run_podman untag $IMAGE $newtag $newtag2
 }
 
+# Regression test for issue #8558
+@test "podman run on untagged image: make sure that image metadata is set" {
+    run_podman inspect $IMAGE --format "{{.ID}}"
+    imageID="$output"
+
+    run_podman untag $IMAGE
+    run_podman run --rm $imageID ls
```

Comment:
I find it helpful in regression tests to include the problem being fixed/tested. For example:
```bash
    # check exit status only; prior to #8623 this would error out with:
    # Error: both RootfsImageName and RootfsImageID must be set if either is set: invalid argument
```
First: should the error ever recur, this would help a maintainer understand how to fix it.

Second: it really isn't clear what this test is doing, which could lead to some future maintainer "helpfully" refactoring it or changing the invocation in some subtle way such that the test would no longer catch the original problem. A comment helps future maintainers understand that the code should not be touched.

Oh well, merged, too late now - but possibly for the backport, and for future such instances.

Reply:
Thanks for the feedback. I will take care to better document in the future.

Reply:
https://github.com/containers/podman/pull/8627